LuckyDragon/app/src/main/java/com/example/luckydragon/ProfileActivity.java

```diff
@@ -41,30 +28,44 @@ enum Mode {
         ORGANIZER,
         ADMIN,
     }
-
     private User user;
+    private GlobalApp.ROLE role;
+    private ProfileView profileView;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_profile);
         getSupportActionBar().setTitle("Profile");
 
-        // Unpack intent
-        Intent intent = getIntent();
+        // Get user from global app
         user = ((GlobalApp) getApplication()).getUser();
-        String role = intent.getStringExtra("role");
+        role = ((GlobalApp) getApplication()).getRole();
+
+        // Create profile view
+        // initializeView() uses profileView, but it will run before the ProfileView constructor returns so it still thinks ProfileView is null
+        // To fix I set isLoaded to false until after profileView is set
+        // I also had to add notifyObservers() to setIsLoaded() in User
+        user.setIsLoaded(false);
+        profileView = new ProfileView(user, this);
+        user.setIsLoaded(true);
+    }
 
+    public void initializeView() {
         // Set profile info views
         TextView nameView = findViewById(R.id.nameTextView);
         TextView emailView = findViewById(R.id.emailTextView);
         TextView phoneNumberView = findViewById(R.id.phoneNumberTextView);
         ImageView profilePictureView = findViewById(R.id.profilePicture);
 
-        nameView.setText(user.getName());
-        emailView.setText(user.getEmail());
-        phoneNumberView.setText(user.getPhoneNumber());
-        profilePictureView.setImageBitmap(user.getProfilePicture());
+        Log.e("Profile", profileView == null ? "null" : profileView.toString());
+        if(profileView != null) {
+            profileView.setName(nameView);
+            profileView.setEmail(emailView);
+            profileView.setPhoneNumber(phoneNumberView);
+            profileView.setProfilePicture(profilePictureView);
+        }
+
```

Comment:
Can't this method be the update method for the view? It also solves the weirdness for setIsLoaded at the top. See i think EventActivity and EventView for an example